
<launch>
	<arg name="compressed" default="true"/>
	<arg name="stereo" default="true"/>
	<arg name="point_cloud_topic" default="/camera/zed_nodelet/mapping/fused_cloud"/>
	<node pkg="wallie_one" type="zed_sub_stereo" name="zed_stereo_node" output="screen" >
		 <param name="stereo_namespace" type="string" value="/camera/zed_nodelet/"/>
   </node>

<node pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" name="depthimage_to_laserscan">
        <remap from="image" to="/camera/zed_nodelet/depth/depth_registered"/>
        <remap from="camera_info" to="/camera/zed_nodelet/depth/camera_info"/>
        <param name="output_frame_id" type="string" value="zed_camera_center"/>
        <param name="use_sim_time" type="bool" value="false"/>
</node>


	<node pkg="wallie_one" type="pcl_reader_no_imagew" name="pcl_reader_no_imagew" output="screen" launch-prefix="xterm -e gdb --args" >
        <param name="use_sim_time" type="bool" value="false"/>

   </node> 
	<node pkg="wallie_one" type="pcl_buffer" name="pcl_buffer_node" output="screen" >
        <param name="use_sim_time" type="bool" value="false"/>

		<param name="point_cloud_topic" type="string" value="$(arg point_cloud_topic)" />

	   </node> 
   	<node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure_node" output="screen" >
        <param name="use_sim_time" type="bool" value="false"/>

   </node> 
 	<node pkg="wallie_one" type="pcl_transformer.py" name="blank_image_pub" output="screen">
        <param name="use_sim_time" type="bool" value="false"/>


   </node> 
 	<node pkg="wallie_one" type="TFBroadcaster.py" name="hmd_transform" output="screen">
        <param name="use_sim_time" type="bool" value="false"/>


   </node>
 	<node pkg="joy" type="joy_node" name="joy_node" output="screen">
        <param name="use_sim_time" type="bool" value="false"/>
		<arg name="autorepeat_rate" value="5.0" />

   </node>
  	<node pkg="multi_object_tracking_lidar" type="kf_tracker" name="kf_tracking" output="screen" >
        <param name="use_sim_time" type="bool" value="false"/>

	<remap from="/pc_roi" to="/filtered_cloud"/>
   </node> 
	<include file="$(find wallie_one)/launch/move_base_zed.launch">
        <param name="use_sim_time" type="bool" value="false"/>
	</include>

	
</launch>
